### PR TITLE
peppy init and progress bar

### LINKED ISSUE #80: `yaml.scanner.ScannerError` when initializing PEP downloaded with `geofetch`
I was trying to initialize a PEP I got from GEO using `geofetch` and `peppy` threw this error: `yaml.scanner.ScannerError: mapping values are not allowed here`

Here is the project config:

```yaml
# Autogenerated by geofetch

name: GSE96155
pep_version: 2.1.0
sample_table: GSE96155_annotation.csv
subsample_table: null

looper:
  output_dir: GSE96155
  pipeline_interfaces: {pipeline_interfaces}

sample_modifiers:
  append:
    Sample_extract_protocol_ch1: general protocol: https://www.encodeproject.org/documents/be2a0f12-af38-430c-8f2d-57953baab5f5/@@download/attachment/Epigenomics_Alternative_Mag_Bead_ChIP_Protocol_v1.1_exp.pdf
    Sample_characteristics_ch1: link: ENCODE dbxrefs Cellosaurus CVCL_B260; http://web.expasy.org/cellosaurus/CVCL_B260
    Sample_data_processing: See GSM*_README.txt supplementary file linked below
    SRR_files: SRA
    
  derive:
    attributes: [read1, read2, SRR_files]
    sources:
      SRA: "${SRABAM}/{SRR}.bam"
      FQ: "${SRAFQ}/{SRR}.fastq.gz"
      FQ1: "${SRAFQ}/{SRR}_1.fastq.gz"
      FQ2: "${SRAFQ}/{SRR}_2.fastq.gz"      
  imply:
    - if: 
        organism: "Mus musculus"
      then:
        genome: mm10
    - if: 
        organism: "Homo sapiens"
      then:
        genome: hg38          
    - if: 
        read_type: "PAIRED"
      then:
        read1: FQ1
        read2: FQ2          
    - if: 
        read_type: "SINGLE"
      then:
        read1: FQ1

project_modifiers:
  amend:
    sra_convert:
      looper:
        results_subdir: sra_convert_results
      sample_modifiers:
        append:
          SRR_files: SRA
          pipeline_interfaces: ${CODE}/geofetch/pipeline_interface_convert.yaml
        derive:
          attributes: [read1, read2, SRR_files]
          sources:
            SRA: "${SRARAW}/{SRR}.sra"
            FQ: "${SRAFQ}/{SRR}.fastq.gz"
            FQ1: "${SRAFQ}/{SRR}_1.fastq.gz"
            FQ2: "${SRAFQ}/{SRR}_2.fastq.gz"
```

I think this is an issue with the `sample_modifiers` section. If I wrap the `append` modifiers in a string with double quotes, the issue resolves. However, I get a **second** error. 

This time `peppy` throws a `TypeError: Provided argument has to be a List[str] or a str, got 'NoneType'`. I was able to solve this by removing entirely the `subsample_table: null` line. Then the project gets initialized properly.

I think there are two "bugs" here.

1. `geofetch` doesn't account for colons messing with the key-value syntax of `yaml` files.
2. A `subsample_table` key of `null` breaks a `peppy.Project()` instantiation. I am wondering if we should either not include the subsample table path in the config file if it doesn't exist or change `peppy` to permit `null` subsample tables.

### PR DESCRIPTION
**Pull request by Khoroshevskyi**
- Added get_projecct function, that returns peppy project.
- Done small refactoring.
- Fixed #80 , #82 , #73
- Fixed some minor errors
-----
**Most of the refactoring and additional charges and cleaning will be done in next PR**

### CODE REVIEW

File: geofetch/config_template.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@
 name: {project_name}
 pep_version: 2.1.0
 sample_table: {annotation}
-subsample_table: {subannotation}
+{subannotation}
```
Comment (nleroy917) [inline geofetch/config_template.yaml:6]: Sorry, I might be confused but why did we remove `subsample_table:` from this?

File: geofetch/geofetch.py [hunk 2]
```diff
@@ -29,43 +31,51 @@
...
-        processed=True,
-        data_source="samples",
-        filter=None,
-        filter_size=None,
-        geo_folder=".",
-        split_experiments=False,
-        bam_folder="",
-        fq_folder="",
-        sra_folder="",
+        name: str = "",
```
Comment (nleroy917) [inline geofetch/geofetch.py:46]: yay type annotations 🥳

File: geofetch/geofetch.py [hunk 3]
```diff
@@ -29,43 +31,51 @@
...
+        processed: bool = False,
+        data_source: str = "samples",
+        filter: str = None,
+        filter_size: str = None,
+        geo_folder: str = ".",
+        split_experiments: bool = False,
+        bam_folder: str = "",
+        fq_folder: str = "",
+        sra_folder: str = "",
         bam_conversion=False,
```
Comment (nleroy917) [inline geofetch/geofetch.py:66]: `bam_conversion: bool = False` ?

File: geofetch/geofetch.py [hunk 4]
```diff
@@ -158,29 +168,88 @@ def __init__(
...
-        if bam_conversion and not just_metadata and not self.which("samtools"):
+        if bam_conversion and not just_metadata and not self._which("samtools"):
             raise SystemExit("For SAM/BAM processing, samtools should be on PATH.")
 
-    def fetch_all(self, input, name=None):
+        self.just_object = False
+
+    def get_project(
+        self, input: str, just_metadata: bool = True, discard_soft: bool = True
+    ) -> Dict[peppy.Project, peppy.Project]:
```
Comment (nleroy917) [inline geofetch/geofetch.py:183]: Is this the correct type annotation for the return object? `Dict[peppy.Project, peppy.Project]` indicates a `dict` whose keys are of the type `peppy.Project`. Did you mean `Dict[str, peppy.Project]`?

File: geofetch/config_template.yaml (hunk 1 above)
Comment (nleroy917) [inline geofetch/config_template.yaml:6]: Why'd we get rid of `subsample_table: `?

**Review by nleroy917 — COMMENTED** [review]
there were a lot of diffs, so I might have missed a few things. It looks good, I had some type annotation things, but I like most of the changes around the API and naming conventions.

File: geofetch/config_template.yaml (hunk 1 above)
Comment (Khoroshevskyi) [reply geofetch/config_template.yaml:6]: I didn't! I have changed it so subsample_table won't be in config file if it doesn't exists. check here: https://github.com/pepkit/geofetch/blob/f4d5adabc5e395afd2ea2e1eac34ade9e6dc2f0c/geofetch/geofetch.py#L1067-L1072

File: geofetch/geofetch.py (hunk 4 above)
Comment (Khoroshevskyi) [reply geofetch/geofetch.py:183]: oops, it should be Dict[str, peppy.Project]. I will change it

**Comment by Khoroshevskyi** [comment]
> there were a lot of diffs, so I might have missed a few things. It looks good, I had some type annotation things, but I like most of the changes around the API and naming conventions.

There isn't too many changes. Bu I have changed the mostly all method names, that's why it looks so bad..

**Review by nleroy917 — APPROVED** [review]

**Pull request merged**